src/Pact/Typechecker.hs

```diff
@@ -805,17 +859,43 @@ toAST (TApp Term.App{..} _) = do
                 return $ over aAppArgs (++ [Var freshArg]) argAST'
             (TyFun t,_) -> die'' argAST $ "App required for funtype argument: " ++ show t
             _ -> return argAST
+
+      -- other special forms: bindings, yield/resume
       case special of
+
         Nothing -> mkApp fun' args'
+
         Just sf -> do
+
           let specialBind = do
-                args'' <- notEmpty _appInfo "Expected >1 arg" (init args')
-                mkApp (set fSpecial (Just (sf,SBinding (last args'))) fun') args''
+                initArgs <- init <$> notEmpty _appInfo "Invalid binding invocation" args'
```

Comment:
Is my understanding correct that previously all bindings required at least one argument (plus the binding), but now we have `resume`, which takes no arguments (plus a binding)?

Reply:
Yep. Seems to do no harm as the other binding funs have differing arity anyways (e.g., `with-default-read`)